apps/molgenis-viz/src/app-components/settings_new_chart.vue

```diff
@@ -0,0 +1,482 @@
+    (row: ColumnSchema) => {
+      return emxTypes.categorical.includes(row.columnType);
+    }
+  );
+
+  if (chartType.value === "barChart") {
+    xVarColumns.value = continuousTableCols;
+    yVarColumns.value = categoricalTableCols;
+  }
+
+  if (["columnChart", "pieChart"].includes(chartType.value)) {
+    xVarColumns.value = categoricalTableCols;
+    yVarColumns.value = continuousTableCols;
+    groupVarColumns.value = categoricalTableCols;
+  }
+
+  if (chartType.value === "scatterPlot") {
+    xVarColumns.value = continuousTableCols;
+    yVarColumns.value = continuousTableCols;
+  }
+}
+
+watch(tableName, (name: string) => {
+  if (name) {
+    const table = props.tables.filter((tbl) => tbl.name === name)[0];
+    selectedTable.value = table;
+    setColumnInputs();
+  }
+});
+
+watch(xVar, (value: string) => {
+  if (value) {
+    xVarType.value = getVarColumnType(value);
+    xVarClass.value = emxTypeAsDataClass(xVarType.value);
+  }
+});
+
+watch(yVar, (value: string) => {
+  if (value) {
+    yVarType.value = getVarColumnType(value);
+    yVarClass.value = emxTypeAsDataClass(yVarType.value);
+  }
+});
+
+watch(groupVar, (value: string) => {
+  if (value) {
+    groupVarType.value = getVarColumnType(value);
+    groupVarClass.value = emxTypeAsDataClass(groupVarType.value);
+  }
+});
+
+watch([chartType, tableName, chartId], (values: Array<String>) => {
+  if (values[0] && values[1]) {
+    chartId.value = `${chartType.value}-${formId.value}`;
+  }
+  if (values[2]) {
+    chartId.value = values[2];
+  }
+});
+
+watch([chartType, tableName, xVar, yVar, groupVar], (params: Array<String>) => {
+  if (params) {
+    const progress = params
+      .map((param: String) => (param ? 1 : 0))
+      .reduce((sum: number, value: number) => sum + value, 0);
+    console.log(progress);
+    showChartPreview.value = progress >= minimalQueryRequirement;
+
+    chartQuery.value = createQuery({
+      table: tableName.value,
+      x: { column: xVar.value, columnType: xVarType.value },
+      y: { column: yVar.value, columnType: yVarType.value },
+      group: { column: groupVar.value, columnType: groupVarType.value },
+    });
+  }
+});
+
+//   chartConfig,
+//   () => {
+//     const currentProgress: Number = Object.keys(chartConfig.value)
+//       .map((key: string) => {
+//         return chartConfig.value[key] !== "" ? 1 : 0;
+//       })
+//       .reduce((sum, value) => sum + value, 0);
+//     showChartPreview.value = currentProgress >= showChartThreshold.value;
+
+//     chartQuery.value = createQuery({
+//       table: chartConfig.value.table,
+//       x: { column: chartConfig.value.x, columnType: chartConfig.value.x_type },
+//       y: { column: chartConfig.value.y, columnType: chartConfig.value.y_type },
+//     });
+//   },
+//   { deep: true }
+// );
+
+// watch(showChartPreview, (status: Boolean) => {
+//   if (status) {
+//     console.log("fetching data....");
+//   }
+// });
```

Comment:
suggest to remove instead of comment out

Reply:
Removed